cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -0,0 +1,485 @@
+            sources=[
+                NodeOrEdgeData(source=source, properties=space_properties)  # type: ignore[arg-type]
+                for space_properties in self.properties.values()
+            ]
+            if self.properties
+            else None,
+        )
+
+
+class InstanceApplyResult(InstanceCore):
+    """A node or edge. This represents the update on the instance.
+
+    Args:
+        instance_type (Literal["node", "edge"]) The type of instance.
+        space (str): The workspace for the instance.a unique identifier for the space.
+        external_id (str): Combined with the space is the unique identifier of the instance.
+        version (str): DMS version of the instance.
+        was_modified (bool): Whether the instance was modified by the ingestion.
+        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+    """
+
+    def __init__(
+        self,
+        instance_type: Literal["node", "edge"],
+        space: str,
+        external_id: str,
+        version: str,
+        was_modified: bool,
+        last_updated_time: int = None,
+        created_time: int = None,
+        **_: dict,
+    ):
+        super().__init__(space, external_id, instance_type)
+        self.version = version
+        self.was_modified = was_modified
+        self.last_updated_time = last_updated_time
+        self.created_time = created_time
+
+
+class NodeApply(InstanceApply):
+    """A node. This is the write version of the node.
+
+    Args:
+        space (str): The workspace for the node.a unique identifier for the space.
+        external_id (str): Combined with the space is the unique identifier of the node.
+        existing_version (int): Fail the ingestion request if the node's version is greater than or equal to this value.
+                                If no existingVersion is specified, the ingestion will always overwrite any
+                                existing data for the edge (for the specified container or node). If existingVersion is
+                                set to 0, the upsert will behave as an insert, so it will fail the bulk if the
+                                item already exists. If skipOnVersionConflict is set on the ingestion request,
+                                then the item will be skipped instead of failing the ingestion request.
+        sources (list[NodeOrEdgeData]): List of source properties to write. The properties are from the node and/or
+                        container the container(s) making up this node.
+    """
+
+    def __init__(
+        self,
+        space: str,
+        external_id: str,
+        existing_version: int = None,
+        sources: list[NodeOrEdgeData] = None,
+        **_: dict,
```

Comment:
Why do we have these? They make it much easier to introduce bugs by e.g. passing "exsiting_version=1"

Reply:
I see we have them for the schema data classes as well. Let's try to get rid of them.

Reply:
We can and should find a way to avoid that without swallowing kwargs

Reply:
as agreed, only doing it on the read classes now.